protostar/commands/test/fuzzing/fuzz_test_case_result_decorator.py

```diff
@@ -0,0 +1,50 @@
+from typing import Optional
+
+from protostar.commands.test.fuzzing.fuzz_input_exception_metadata import (
+    FuzzInputExceptionMetadata,
+)
+from protostar.commands.test.test_environment_exceptions import ReportedException
+from protostar.commands.test.test_results import (
+    FailedFuzzTestCaseResult,
+    FailedTestCaseResult,
+    FuzzResult,
+    FuzzTestExecutionResult,
+    PassedFuzzTestCaseResult,
+    PassedTestCaseResult,
+    TestCaseResultDecorator,
+)
+
+
+class FuzzTestCaseResultDecorator(TestCaseResultDecorator):
+    def decorate_passed(
+        self,
+        result: PassedTestCaseResult,
+        execution_result: FuzzTestExecutionResult,
```

Comment:
pyright should've scream here...

When `reportIncompatibleMethodOverride` the following error is thrown:
```
[REDACTED]/protostar/commands/test/fuzzing/fuzz_test_case_result_decorator.py
  [REDACTED]/protostar/commands/test/fuzzing/fuzz_test_case_result_decorator.py:19:9 - error: Method "decorate_passed" overrides class "TestCaseResultDecorator" in an incompatible manner
    Parameter 3 type mismatch: base parameter is type "TestExecutionResult", override parameter is type "FuzzTestExecutionResult"
      "TestExecutionResult" is incompatible with "FuzzTestExecutionResult" (reportIncompatibleMethodOverride)
```

(Btw. we need to enable this check.)

Reply:
Done.